torch/utils/cpp_extension.py

```diff
@@ -312,14 +312,16 @@ class BuildExtension(build_ext, object):
     @classmethod
     def with_options(cls, **options):
         r'''
-        Returns an alternative constructor that extends any original keyword
+        Returns a subclass with alternative constructor that extends any original keyword
         arguments to the original constructor with the given options.
         '''
-        def init_with_options(*args, **kwargs):
-            kwargs = kwargs.copy()
-            kwargs.update(options)
-            return cls(*args, **kwargs)
-        return init_with_options
+        class cls_with_options(cls):
+            def __init__(self, *args, **kwargs):
+                kwargs = kwargs.copy()
```

Comment:
I don't think the copy is necessary, you'll get a fresh kwargs dict every time this is called.